janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/VertexIDAssignerTest.java

```diff
@@ -31,58 +31,66 @@
 import org.janusgraph.graphdb.internal.InternalRelation;
 import org.janusgraph.graphdb.internal.InternalVertex;
 
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import static org.junit.Assert.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.*;
 
 import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.List;
+import java.util.stream.Stream;
 
 /**
  * @author Matthias Broecheler (me@matthiasb.com)
  */
-
-@RunWith(Parameterized.class)
 public class VertexIDAssignerTest {
 
-    final VertexIDAssigner idAssigner;
+    private VertexIDAssigner idAssigner;
 
-    @Parameterized.Parameters
-    public static Collection<Object[]> configs() {
-        final List<Object[]> configurations = new ArrayList<>();
+    public static Stream<Arguments> configs() {
```

Comment:
If it is OK I would rewrite it to something like this (not tested):
```
public static Stream<Arguments> configs() {
    final List<Arguments> configurations = new ArrayList<>();

    for (int maxPerPartition : new int[]{Integer.MAX_VALUE, 100, 300}) {
        for (int numPartitions : new int[]{2, 4, 10}) {
            for (int[] local : new int[][]{null, {0,2, numPartitions}, {235,234,8}, {1,1,2}, {0,1<<(numPartitions-1),numPartitions}}) {
                configurations.add(Arguments.arguments(generateConfigurationArguments(numPartitions, maxPerPartition, local)));
            }
        }
    }

    return configurations.stream();
}

private static Arguments generateConfigurationArguments(int numPartitionsBits, int partitionMax, int[] localPartitionDef){
    MockIDAuthority idAuthority = new MockIDAuthority(11, partitionMax);

    StandardStoreFeatures.Builder fb = new StandardStoreFeatures.Builder();

    if (null != localPartitionDef) {
        fb.localKeyPartition(true);
        idAuthority.setLocalPartition(PartitionIDRangeTest.convert(localPartitionDef[0],localPartitionDef[1],localPartitionDef[2]));
    }
    StoreFeatures features = fb.build();

    ModifiableConfiguration config = GraphDatabaseConfiguration.buildGraphConfiguration();
    config.set(GraphDatabaseConfiguration.CLUSTER_MAX_PARTITIONS,1<<numPartitionsBits);
    idAssigner = new VertexIDAssigner(config, idAuthority, features);
    System.out.println(String.format("Configuration [%s|%s|%s]",numPartitionsBits,partitionMax,Arrays.toString(localPartitionDef)));

    long maxIDAssignments;
    if (localPartitionDef!=null && localPartitionDef[0]<localPartitionDef[1] && localPartitionDef[2]<=numPartitionsBits) {
        maxIDAssignments = ((localPartitionDef[1]-localPartitionDef[0])<<(numPartitionsBits-localPartitionDef[2]))*((long)partitionMax);
    } else {
        maxIDAssignments = (1<<numPartitionsBits)*((long)partitionMax);
    }

    return Arguments.arguments(idAssigner, maxIDAssignments, numPartitionsBits);
}

private void setup(VertexIDAssigner idAssigner, long maxIDAssignments, int numPartitionsBits){
    this.idAssigner = idAssigner;
    this.maxIDAssignments = maxIDAssignments;
    this.numPartitionsBits = numPartitionsBits;
}
```

In such case your `setup` method will be lightweight. Even so, we can completely remove `setup` method and use method variables in tests instead of class variables.

Reply:
Tests will look something like this if we use class variables:
```
@ParameterizedTest
@MethodSource("configs")
public void testIDAssignment(VertexIDAssigner idAssigner, long maxIDAssignments, int numPartitionsBits) {
    setup(idAssigner, maxIDAssignments, numPartitionsBits);
   // Actual test below
}
```
or something like this if we use method variables:
```
@ParameterizedTest
@MethodSource("configs")
public void testIDAssignment(VertexIDAssigner idAssigner, long maxIDAssignments, int numPartitionsBits) {
   // Actual test below
}
```